### PR TITLE
Use spotinst provider

### PR DESCRIPTION
**Pull request by ImpSy**

### CODE REVIEW

File: examples/gcp-from-scratch/main.tf [hunk 1]
```diff
@@ -84,7 +84,7 @@ provider "spotinst" {
 }
 
 resource "spotinst_ocean_gke_import" "ocean" {
-  cluster_name          = var.cluster_name
+  cluster_name          = google_container_cluster.cluster.name
```
Comment (Z4ck404) [inline examples/gcp-from-scratch/main.tf:87]: shouldn't we add a `depends_on` as well in this case.

Comment (Z4ck404) [reply examples/gcp-from-scratch/main.tf:87]: please ignore that comment 😅 Just checked again, it's not necessary in this case. the dependency should work without it.

File: main.tf [hunk 2]
```diff
@@ -36,36 +36,24 @@ resource "kubernetes_cluster_role_binding" "deployer" {
   }
 }
 
-resource "kubernetes_job" "deployer" {
-  count = var.create_cluster ? 1 : 0
-  metadata {
-    generate_name = "ofas-deploy-"
-    namespace     = local.spot_system_namespace
+resource "spotinst_ocean_spark" "cluster" {
```
Comment (Z4ck404) [inline main.tf:39]: 🚀

Comment (Z4ck404) [reply main.tf:39]: Managed to create a gcp from scratch

**Review by Z4ck404 — APPROVED** [review]

File: main.tf (hunk 2 above)
Comment (Z4ck404) [reply main.tf:39]: @ImpSy  I will do some tests today. so we can merge if everything goes well  : 

Examples tested : 

- [ ] from privtae vpc
- [ ] from scratch 
- [x] gcp from scratch 
- [ ] gcp from vpc 
- [ ] gcp import gke cluster
- [ ] import eks cluster
- [ ] import ocean cluster

Comment (Z4ck404) [reply main.tf:39]: @ImpSy the destroy is working as expected though 
Screenshot 2022-10-07 at 12 58 37

File: variables.tf [hunk 3]
```diff
@@ -1,23 +1,46 @@
...
   default     = true
 }
 
-variable "deployer_image" {
-  type        = string
-  description = "Specifies the Docker image name used in the deployer Job"
-  default     = "public.ecr.aws/f4k1p1n4/bigdata-deployer"
+variable "compute_create_vngs" {
+  type        = bool
+  description = "Controls whether the Ofas cluster should create its dedicated Vngs"
```
Comment (thorsteinnth) [inline variables.tf:14]: Controls whether dedicated Ocean Spark VNGs will be created by the cluster creation process

File: variables.tf [hunk 4]
```diff
@@ -1,23 +1,46 @@
...
+  default     = true
 }
 
-variable "deployer_tag" {
-  type        = string
-  description = "Specifies the Docker image tag used in the deployer Job"
-  default     = "main"
+variable "compute_use_taints" {
+  type        = bool
+  description = "Controls whether the Ofas cluster will use taints to schedule workloads"
```
Comment (thorsteinnth) [inline variables.tf:20]: Controls whether the Ocean Spark cluster will use taints to schedule workloads

File: variables.tf [hunk 5]
```diff
@@ -1,23 +1,46 @@
...
+  default     = true
 }
 
-variable "image_pull_policy" {
-  type        = string
-  description = "Specifies the image pull policy (one of: Always, Never, IfNotPresent)"
-  default     = "Always"
+variable "ingress_service_annotations" {
+  type        = map(string)
+  description = "Allow custom service annotations that will be the ingress for customization"
```
Comment (thorsteinnth) [inline variables.tf:26]: Annotations that will be added to the load balancer service, allowing for customization of the load balancer

File: variables.tf [hunk 6]
```diff
@@ -1,23 +1,46 @@
...
+  default     = {}
+}
+
+variable "log_collection_collect_driver_logs" {
+  type        = bool
+  description = "Controls whether the Ofas cluster will collect Spark driver logs"
```
Comment (thorsteinnth) [inline variables.tf:32]: Ofas -> Ocean Spark

File: variables.tf [hunk 7]
```diff
@@ -1,23 +1,46 @@
...
+  default     = true
+}
+
+variable "webhook_use_host_network" {
+  type        = bool
+  description = "Controls whether Ofas system pods will use host network"
```
Comment (thorsteinnth) [inline variables.tf:38]: Controls whether Ocean Spark system pods that expose webhooks will use the host network

File: examples/import-eks-cluster/main.tf [hunk 8]
```diff
@@ -45,7 +45,7 @@ module "ocean-controller" {
   spotinst_token   = var.spotinst_token
   spotinst_account = var.spotinst_account
 
-  cluster_identifier = var.cluster_name
+  cluster_identifier = module.ocean-aws-k8s.ocean_id
```
Comment (thorsteinnth) [inline examples/import-eks-cluster/main.tf:48]: is this correct? this should be the controller cluster ID?

Comment (Z4ck404) [reply examples/import-eks-cluster/main.tf:48]: the ocean_id is the only output this resource has 
https://registry.terraform.io/modules/spotinst/ocean-aws-k8s/spotinst/latest?tab=outputs

Comment (Z4ck404) [reply examples/import-eks-cluster/main.tf:48]: Screenshot 2022-10-12 at 14 15 11

@thorsteinnth  from the tfstate file of the test we did with Seb, it's the same as cluster name and controller cluster id, for some reason

**Review by thorsteinnth — APPROVED** [review]

**Pull request merged**